server/mentionscrawler/blueprints/company.py

```diff
@@ -15,10 +18,34 @@ def update_companies(user):
     names = request.get_json()
     if len(names) == 0:
         return bad_request_response("Must have at least one company name")
-    Company.query.filter_by(mention_user_id=user.get("user_id")).delete()
+    old_companies = Company.query.filter_by(mention_user_id=user.get("user_id")).all()
+    old_companies_map = {}
+    old_mentions = []
     companies = []
+
+    count = 0
+    for company in old_companies:
+        old_companies_map[company.name] = count
+        count = count + 1
+
     for name in names:
-        companies.append(Company(user.get("user_id"), name))
-    insert_rows(companies)
-    return ok_response("Company names updated!")
+        if name in old_companies_map:  # Spare existing names from deletion
+            del old_companies[old_companies_map[name]]
+        company_count = Company.query.filter_by(name=name).count()
+        if company_count == 0:  # ensure existing names don't try to get inserted
+            companies.append(Company(user.get("user_id"), name))
+
+    for company in old_companies:  # Line up mentions for execution
+        old_mentions = old_mentions + Mention.query.filter_by(mention_user_id=user.get("user_id"),
+                                                              company=company.id).all()
 
+    result = delete_rows(old_mentions)
+    if result is not True:
+        return result
+    result = delete_rows(old_companies)
+    if result is not True:
+        return result
+    result = insert_rows(companies)
+    if result is not True:
+        return result
```

Comment:
possibly but I feel like I'll need to find a better way to explain this since it will be a common practice unless I can think of a better way to achieve the same goal

Basically the database functions will return True if the database operation was a successful, or will return an http response if it failed, so I needed a way of returning the http method to the front end, and aside from having a global state system that sends off http results (which I'm not even sure how that would work in python) I'm not entirely sure how else to handle this...

This would be the perfect use scenario for the maybe type I was telling you about, in haskell...